lib/redisStore.js

```diff
@@ -0,0 +1,52 @@
+const createKey = (key, namespace) => {
+  return namespace + key
+}
+
+const REDIS_KEY_TTL_IS_NOT_SET = -1
+export class RedisStore {
```

Comment:
I'm not sure where this implementation came from exactly, but I'm pretty sure this is not a thread-safe implementation as it doesn't use atomic primitives in redis or locks. it may be good enough for our purposes, but it's certainly not a correct implementation of a rate limiter. is it inspired from an existing implementation? unfortunately I'm not familiar enough with the right way to do it in redis to suggest an implementation.

I'm also not entirely sure if this is really an issue since due to the way it's being used, it will never bump into race conditions (as the most common scenario would be that the app is running on a single thread), but maybe something to keep in mind for the future. 

If you look at the redis docs, you'll see that getting a rate limiter right is quite a bit of work.

https://redis.io/commands/incr/ (scroll down)

Reply:
I've taken this implementation from the `fastify-rate-limit` plugin but they don't implement decrement method so ...